lib/rdf/model/uri.rb

```diff
@@ -739,7 +739,7 @@ def ==(other)
         # If other is a Literal, reverse test to consolodate complex type checking logic
         other == self
       when String then to_s == other
-      when URI then hash == other.hash && to_s == other.to_s
+      when URI then hash == other.hash
       else other.respond_to?(:to_uri) && to_s == other.to_uri.to_s
```

Comment:
that was my thought too, but I didn't want to change too much at one time. Maybe a future ticket?